src/source/Ice/SocketConnection.c

```diff
@@ -197,7 +199,25 @@ STATUS socketConnectionSendData(PSocketConnection pSocketConnection, PBYTE pBuf,
         }
 
         // sending through UDP never block
-        if (sendto(pSocketConnection->localSocket, pBuf, bufLen, 0, destAddr, addrLen) < 0) {
+        result = sendto(pSocketConnection->localSocket, pBuf, bufLen, 0, destAddr, addrLen);
+        if (result < 0 && (errno == EAGAIN || errno == EWOULDBLOCK)) {
```

Comment:
Combine this if and the next into one

if (result < 0 ) {
  switch (errno) {
case foo:
// Explicit fall-through
case bar:
// handle it
break;
default:
warn();
}
}

Reply:
EWOULDBLOCK and EAGAIN are the same value and cant be put in a switch state. These two are the only errno we gonna handle.

Reply:
Perhaps those are platform specific? At any rate, you can still do if/else but combine the outer if together